reconfiguration/src/reconfiguration_handler.cpp

```diff
@@ -63,50 +63,40 @@ bool ReconfigurationHandler::handle(const concord::messages::AddRemoveWithWedgeC
                                     concord::messages::ReconfigurationResponse&) {
   LOG_INFO(getLogger(), "AddRemoveWithWedgeCommand instructs replica to stop at seq_num " << bft_seq_num);
   bftEngine::ControlStateManager::instance().setStopAtNextCheckpoint(bft_seq_num);
-  bftEngine::ControlStateManager::instance().setRestartBftFlag(command.bft_support);
-  if (command.bft_support) {
-    bftEngine::IControlHandler::instance()->addOnStableCheckpointCallBack([=]() {
-      bftEngine::ControlStateManager::instance().markRemoveMetadata();
-      bftEngine::EpochManager::instance().setNewEpochFlag(true);
-    });
-    if (command.restart) {
+  handleWedgeCommands(command.bft_support, true, command.restart, command.restart);
+  return true;
+}
+void ReconfigurationHandler::handleWedgeCommands(bool bft_support, bool remove_metadata, bool restart, bool unwedge) {
+  if (unwedge) bftEngine::ControlStateManager::instance().setRestartBftFlag(bft_support);
+  if (bft_support) {
+    if (remove_metadata)
+      bftEngine::IControlHandler::instance()->addOnStableCheckpointCallBack(
+          [=]() { bftEngine::ControlStateManager::instance().markRemoveMetadata(); });
+    if (restart)
       bftEngine::IControlHandler::instance()->addOnStableCheckpointCallBack(
           [=]() { bftEngine::ControlStateManager::instance().sendRestartReadyToAllReplica(); });
-    }
+    if (unwedge)
+      bftEngine::IControlHandler::instance()->addOnStableCheckpointCallBack(
+          [=]() { bftEngine::EpochManager::instance().setNewEpochFlag(true); });
   } else {
-    bftEngine::IControlHandler::instance()->addOnSuperStableCheckpointCallBack([=]() {
-      bftEngine::ControlStateManager::instance().markRemoveMetadata();
-      bftEngine::EpochManager::instance().setNewEpochFlag(true);
-    });
-    if (command.restart) {
+    if (remove_metadata)
+      bftEngine::IControlHandler::instance()->addOnStableCheckpointCallBack(
+          [=]() { bftEngine::ControlStateManager::instance().markRemoveMetadata(); });
+    if (restart)
       bftEngine::IControlHandler::instance()->addOnSuperStableCheckpointCallBack(
           [=]() { bftEngine::ControlStateManager::instance().sendRestartReadyToAllReplica(); });
-    }
+    if (unwedge)
+      bftEngine::IControlHandler::instance()->addOnStableCheckpointCallBack(
+          [=]() { bftEngine::EpochManager::instance().setNewEpochFlag(true); });
   }
-  return true;
 }
 
 bool ReconfigurationHandler::handle(const concord::messages::RestartCommand& command,
```

Comment:
concord::messages::RestartCommand with restart=False is same as wedge command. Do we need restart flag for this command? is there a use case where we send RestartCommand with restart=False?